### PR TITLE
Docs: warn `parallelism > 1` doesn't work for migration procedures

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Review by jbonofre — APPROVED** [review]
It looks reasonable to me to warn the users waiting the actual fix.

File: docs/docs/spark-procedures.md [hunk 1]
```diff
@@ -629,6 +632,9 @@ By default, the original table is retained with the name `table_BACKUP_`.
 | `backup_table_name` |  | string | Name of the table that will be retained as backup (defaults to `table_BACKUP_`) |
 | `parallelism` |   | int | Number of threads to use for file reading (defaults to 1) |
 
+!!! warning
+    There's a [known issue with `parallelism > 1`](https://github.com/apache/iceberg/issues/11147). We will fix it in next release.
```
Comment (Fokko) [inline docs/docs/spark-procedures.md:636]:
```suggestion
    There's a [known issue with `parallelism > 1`](https://github.com/apache/iceberg/issues/11147) that is scheduled to be fixed in the next release.
```

File: docs/docs/spark-procedures.md [hunk 2]
```diff
@@ -675,6 +681,9 @@ Warning : Schema is not validated, adding files with different schema to the Ice
 
 Warning : Files added by this method can be physically deleted by Iceberg operations
 
+!!! warning
+    There's a [known issue with `parallelism > 1`](https://github.com/apache/iceberg/issues/11147). We will fix it in next release.
```
Comment (Fokko) [inline docs/docs/spark-procedures.md:685]:
```suggestion
    There's a [known issue with `parallelism > 1`](https://github.com/apache/iceberg/issues/11147) that is scheduled to be fixed in the next release.
```

**Review by Fokko — COMMENTED** [review]
I think it is fair to fix this down the line. One small nit on the wording, I try to avoid `we`.

**Comment by RussellSpitzer** [comment]
Thanks @manuzhang We'll figure out that fix before the next release I promise. Thanks @Fokko for review

**Pull request merged**